### DISCUSSION TITLE
Could not get connection while getPartitionedTopicMetadata - Connection handshake failed

### DISCUSSION THREAD

**Discussion by championii** [apache/pulsar#20838] [Q&A]
I have setup apache pulsar by following this tutorial: https://pulsar.apache.org/docs/3.0.x/deploy-aws/ but I use aws ansible dynamic inventory rather than terraform inventory in a tutorial. The problem is that once I run the following code

import org.apache.pulsar.client.api.*;

public class test {
    public static void main(String[] args) throws PulsarClientException {
        PulsarClient client = PulsarClient.builder()
            .serviceUrl(<my pulsar web url>) 
            .build();
        
        Producer<String> producer = client.newProducer(Schema.STRING)
        .topic("my-topic")
        .create();

        producer.newMessage()
        .value("HelloWorld")
        .send();

        producer.close();
        client.close();
        
    }    
}
the following error pops up:

10:17:35.419 [pulsar-client-io-1-7] DEBUG org.apache.pulsar.client.impl.ClientCnx - [id: 0x7b3029f5, L:/192.168.1.78:52743 - R:] Connected to broker 10:17:35.421 [pulsar-client-io-1-7] DEBUG org.apache.pulsar.client.impl.ClientCnx - Complete: true 10:17:35.604 [pulsar-client-io-1-7] DEBUG org.apache.pulsar.client.impl.ConnectionPool - Removing closed connection from pool: AbstractChannel$CloseFuture@3aac5716(success) 10:17:35.604 [pulsar-client-io-1-7] INFO org.apache.pulsar.client.impl.ClientCnx - [id: 0x7b3029f5, L:/192.168.1.78:52743 ! R:] Disconnected 10:17:35.605 [pulsar-client-io-1-7] WARN org.apache.pulsar.client.impl.ConnectionPool - [[id: 0x7b3029f5, L:/192.168.1.78:52743 ! R:]] Connection handshake failed: org.apache.pulsar.client.api.PulsarClientException: Connection already closed 10:17:35.994 [pulsar-client-scheduled-30-1] WARN org.apache.pulsar.client.impl.PulsarClientImpl - [topic: persistent://public/default/test-topic] Could not get connection while getPartitionedTopicMetadata -- Will try again in 384 ms

and these block of message repeats forever. I intentionally remove/hide my pulsar web URL for privacy purposes. I tried to disable my firewall but it does not resolve the error. Does anyone know where the issues are and how I can fix them? I am pretty new to apache pulsar so thank you in advance...

**Comment by thomasechen** [comment]
Hi  [championii](https://github.com/championii)

For the code your provided. Did you provide the URL into the parameter serviceURL ???

PulsarClient client = PulsarClient.builder()
.serviceUrl( pular://hostname:6650 )
.build();

**Reply by championii** [reply]
for privacy purpose, I prefer to keep my serviceURL a secret :). Yeah, I really struggle to resolve this error? do you have any idea?

**Comment by thomasechen** [comment]
Hi Championii,

 It coulde be any resaon there . Did u enable the topic auto creation
setting ?

championii ***@***.***> 於 2023年7月24日 週一 22:25 寫道：

**Reply by championii** [reply]
What is topic auto creation setting? It might work...

**Comment by thomasechen** [comment]
There is a setting called allowAutoTopicCreation in the broker
configuration. Do u set the value of it as true?

championii ***@***.***> 於 2023年7月24日 週一 23:11 寫道：
